pkgs/by-name/va/vale-ls/package.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, openssl
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "vale-ls";
+  version = "0.3.7";
+
+  src = fetchFromGitHub {
+    owner = "errata-ai";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-xsYokuRAdZZ3lz9/J02QDPMR3p0pxxqD1ljRdQI5GKE=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
+    pkg-config
+  ];
+
+  buildInputs = [ openssl ];
+
+  # The following tests are reaching to the network.
+  checkFlags = [
+    "--skip=vale::tests"
+  ];
+
+  cargoHash = "sha256-QEU9e8KTAnLD9ykV+C87iVKojX2aEDRwocwvZcxpOU8=";
+  cargoPatches = [ ./0001-Update-dependencies.patch ];
```

Comment:
Instead of patching `Cargo.toml` to remove the `vendored` feature flag from `openssl`, can we set the `OPENSSL_NO_VENDOR` environment variable at build time? Seems easier than maintaining a patch and regenerating the lockfile every time there's a release.

From the [relevant documentation](https://docs.rs/openssl/latest/openssl/#manual):

> `OPENSSL_NO_VENDOR` - If set, always find OpenSSL in the system, even if the vendored feature is enabled.